src/utils/url.ts

```diff
@@ -55,7 +54,11 @@ export const openUrl = (url: string) => {
     // return;
   }
 
-  window.open(url, "_blank", "noopener,noreferrer");
+  if (isExternalUrl(url)) {
+    window.open(url, "_blank", "noopener,noreferrer");
+  } else {
+    window.location.href = url;
+  }
```

Comment:
Can/should we be making use of our routing navigation helpers here? I believe this code as is will force a full window refresh, whereas using the routing helpers will do it within the same 'page load'.

Ideally we would be using the `useHistory` hook, but we're not in a react component context here, so we can't use hooks:

- https://reactrouter.com/web/api/Hooks/usehistory

So I _think_ we can just use `history` directly.. but it seems to come with extra caveats/warnings that i'm not immediately sure of the ramifications of:

- https://reactrouter.com/web/api/history

I think you may want to use `history.push` here, but it would be worth doing your own quick research/testing to validate that:

> `push(path, [state])` - (function) Pushes a new entry onto the history stack

Reply:
If this ends up being too complex, or like it will be a rabbithole of unknown length, feel free to add a `@debt` comment such as `@debt can we use react-router's navigation helpers here? should we?` and then leave this implementation as is (eg. `window.location.href = url`)

Reply:
On one side I understand your point and why we should use history/router, on another I feel like using it outside component logic becomes complicated and harder to trace and I haven't seen it anywhere tbh. Importing history directly would feel like importing redux store directly. I'll leave it as it is for now.

Reply:
I'm happy with leaving it for now.

The reasoning behind it is essentially as a single page application we don't actually want to be navigating to different pages, as what we are really doing is 'faking' route changes within the router/etc itself. That's where using the router helpers comes into play. I expect we haven't done that elsewhere in the app more out of 'tech debt' reasons than 'we explicitly chose to do it this way' reasons.